src/buffer/message_view.rs

```diff
@@ -0,0 +1,312 @@
+use crate::buffer::scroll_view::Message;
+use data::isupport::CaseMap;
+use data::server::Server;
+use data::target::{self};
+use data::{Config, User, message};
+use iced::advanced::text;
+use iced::widget::{column, container, row};
+
+use super::user_context;
+use crate::widget::{
+    Element, message_content, message_marker, selectable_text,
+};
+use crate::{Theme, theme};
+
+pub trait MessageFormat<'a> {
```

Comment:
Wonder if we can find a better name here 🤔

Reply:
maybe `DrawMessage`